src/merkle/store/mod.rs

```diff
@@ -320,16 +320,16 @@ impl<T: KvMap<RpoDigest, StoreNode>> MerkleStore<T> {
         Ok(())
     }
 
-    /// Appends the provided [MerklePathSet] into the store.
+    /// Appends the provided [PartialMerkleTree] into the store.
     ///
     /// For further reference, check [MerkleStore::add_merkle_path].
-    pub fn add_merkle_path_set(
+    pub fn add_partial_merkle_tree(
         &mut self,
-        path_set: &MerklePathSet,
+        partial_mt: &PartialMerkleTree,
     ) -> Result<RpoDigest, MerkleError> {
-        let root = path_set.root();
-        for (index, path) in path_set.to_paths() {
-            self.add_merkle_path(index, path.value, path.path)?;
+        let root = partial_mt.root();
+        for (index, path) in partial_mt.to_paths() {
+            self.add_merkle_path(index.value(), path.value, path.path)?;
         }
         Ok(root)
     }
```

Comment:
I would probably get rid of this and instead implement `From<&PartialMerkleTree>` similar how we do it for other data structures starting on line 410.